From a91d756d9574aeab8a8105ba3bdf92c27c4bd74a Mon Sep 17 00:00:00 2001
From: Ricardo Neri <ricardo.neri-calderon@linux.intel.com>
Date: Thu, 23 Jul 2015 16:29:00 -0700
Subject: [PATCH] cpu/cpufreq: Iterate over the correct number of CPUs in
 _test_sw_any

cpu_test_sw_any goes into a loop to set the frequency of all the
available CPUs. However, the loop iterates beyond the limits of the
CPUs array. Given that functions such as cpu_set_lowest_frequency
derreference members of struct cpu, iterating beyond the limits of the
array might cause segmentation faults.

Upstream status: submitted

Signed-off-by: Ricardo Neri <ricardo.neri-calderon@linux.intel.com>
---
 src/cpu/cpufreq/cpufreq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpu/cpufreq/cpufreq.c b/src/cpu/cpufreq/cpufreq.c
index 40f6a30..35812f4 100644
--- a/src/cpu/cpufreq/cpufreq.c
+++ b/src/cpu/cpufreq/cpufreq.c
@@ -507,7 +507,7 @@ static int cpufreq_test_sw_any(fwts_framework *fw)
 
 	ok = true;
 
-	for (i = 0; i <= num_cpus; i++) {
+	for (i = 0; i < num_cpus; i++) {
 		struct cpu *cpu = &cpus[i];
 
 		if (!cpu->online)
-- 
1.9.1

